src/main/java/seedu/storemando/logic/commands/ListCommand.java

```diff
@@ -14,11 +18,29 @@
 
     public static final String MESSAGE_SUCCESS = "Listed all items";
 
+    private final Predicate<Item> predicate;
+
+    public ListCommand(LocationContainsKeywordsPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    public ListCommand() {
+        this.predicate = PREDICATE_SHOW_ALL_ITEMS;
+    }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredItemList(PREDICATE_SHOW_ALL_ITEMS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        model.updateFilteredItemList(predicate);
```

Comment:
For this when it is showing all items will it also work as normal?

Reply:
Yes, it will work as normal. I am using inheritance to achieve both showing all items when no argument is given and also showing filtered items when arguments are given.